cpp/src/arrow/type.h

```diff
@@ -538,9 +538,11 @@ class ARROW_EXPORT Date64Type : public DateType {
   static std::string name() { return "date"; }
 };
 
-enum class TimeUnit : char { SECOND = 0, MILLI = 1, MICRO = 2, NANO = 3 };
+struct TimeUnit {
+  enum type { SECOND = 0, MILLI = 1, MICRO = 2, NANO = 3 };
+};
```

Comment:
enum class is difficult for tools like Cython to wrap; since we're already using the `struct ... { enum ... ` pattern (more the norm in C++0x) for `arrow::Type`, we might as well use it more consistently